### PR TITLE
bunyan+loggly

### PR DESCRIPTION
**Pull request by cflynn07**
- [ ] merge https://github.com/CodeNow/devops-scripts/pull/84
- [ ] test on staging

### CODE REVIEW

**Comment by cflynn07** [comment]
Deployed to staging
`ansible-playbook -i ./stage-hosts docker-listener.yml -e git_branch=SAN-2205-bunyan-loggly -e redis_host=10.0.1.210 -e rabbit_host=10.0.1.210 -e redis_port=32839 -e rabbit_port=32855 -e loggly_token=f673760d-e0b3-4a93-a15e-2862ea074f91`

**Comment by cflynn07** [comment]
@anandkumarpatel we never fixed deployment of docker-listener to staging docks. Docker-listener isn't running ATM on alpha-stage-dock1

File: test/fixtures/docker-server.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 'use strict';
 var app = require('docker-mock');
-var debug = require('auto-debug')();
 
 app.listen(4243, function (err) {
   if (err) { throw err; }
-  debug('started docker mock');
+  console.log('started docker mock');
```
Comment (bkendall) [inline test/fixtures/docker-server.js:6]: why are these going on `console.log`? we try to avoid this...

File: lib/listener.js [hunk 2]
```diff
@@ -79,31 +79,32 @@ Listener.prototype.stop = function () {
  */
 function isWritable (streamInst, name) {
   if (!streamInst.write) {
-    debug(name + ' is not writable');
+    log.error({
+      name: name
+    }, 'not writable');
```
Comment (rsandor) [inline lib/listener.js:84]: Perhaps be more descriptive? Suggest: `Event stream is not writable`

File: lib/publisher.js [hunk 3]
```diff
@@ -60,18 +60,21 @@ RedisPublisherStream.prototype._write = function (chunk, encoding, callback) {
       case 'create':
         // ignore build containers
         if (isUserContainer(enhanced)) {
-          log.info({enhanced: enhanced}, 'container-create');
+          log.info({
+            enhanced: enhanced
+          }, 'container-create');
```
Comment (rsandor) [inline lib/publisher.js:65]: Perhaps a little better description?

File: main.js [hunk 4]
```diff
@@ -32,12 +33,14 @@ function start (port, cb) {
   cb = cb || noop;
   server = app.listen(port, function (err) {
     if (err) { return cb(err); }
-    debug('server listen on', port);
+    log.info({
+      port: port
+    }, 'server listen');
```
Comment (rsandor) [inline main.js:38]: Rename message to `Server started and listeneing`?

File: test/fixtures/docker-mock.js [hunk 5]
```diff
@@ -1,11 +1,10 @@
 'use strict';
 var spawn = require('child_process').fork;
-var debug = require('auto-debug')();
 
 module.exports.start = function (cb) {
   this.server = spawn('test/fixtures/docker-server.js');
 
-  debug('Spawned child pid: ', this.server.pid);
+  console.log('Spawned child pid: ', this.server.pid);
```
Comment (rsandor) [inline test/fixtures/docker-mock.js:7]: Should we continue using debug here? Console.log doesn't have nearly the same level of control. Even though it is a only local testing message we may want to filter these in the future.

File: test/fixtures/docker-mock.js [hunk 6]
```diff
@@ -16,7 +15,7 @@ module.exports.start = function (cb) {
 module.exports.stop = function (cb) {
   if (this.server) {
     this.server.on('exit', function (code) {
-      debug('docker server process exited with code ', code);
+      console.log('docker server process exited with code ', code);
```
Comment (rsandor) [inline test/fixtures/docker-mock.js:18]: See comment above.

File: test/fixtures/docker-server.js (hunk 1 above)
Comment (rsandor) [inline test/fixtures/docker-server.js:6]: I mentioned considering continue use of debug above.

**Comment by rsandor** [comment]
Similar thoughts to that of the navi logging. I like that you sprinkling logging (esp. the trace logging) in many places, but I kinda want the log lines themselves to a bit more descriptive. Other than that, looks pretty good :)

File: test/fixtures/docker-server.js (hunk 1 above)
Comment (cflynn07) [inline test/fixtures/docker-server.js:6]: what's the problem with console.log in tests? I didn't have the energy to convert to bunyan this weekend. We want to move away from debug and remove debug from npm dependencies.

File: test/fixtures/docker-mock.js (hunk 5 above)
Comment (cflynn07) [inline test/fixtures/docker-mock.js:7]: I really don't want debug in the project / promote it's use. The team should be using bunyan w/ the bunyan cli

Comment (cflynn07) [inline test/fixtures/docker-mock.js:7]: i was lazy here, I had been pruning out the debug statements and converting them to bunyan for hours and I slacked off in the tests and just find&replaced them to console.log

**Comment by cflynn07** [comment]
staging this is blocked on ansbile

**Comment by anandkumarpatel** [comment]
@cflynn07 issue is with your ssh keys. I just fixed the problem on my machine by `ssh-add ~/.ssh/id_rsa`
run `ssh-add -L` and ensure your github rsa key is shown in the list.
also ssh into the server and run `ssh-add -L` and make sure that id_rsa is also in the list. 

https://developer.github.com/guides/using-ssh-agent-forwarding/

**Comment by cflynn07** [comment]
sweet it works.

**Comment by podviaznikov** [comment]
code looks good to me. did you stage it?

**Comment by cflynn07** [comment]
@podviaznikov yeah it was staged

**Pull request merged**